Bottom: b99fc633c6a2295539fa958c41f38acbda1f9873
Top:    04ec74a228db26899dfcbcde337cff4e8237778c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-09 19:27:35 +0000

HID: Make arguments to the prompt_for method const char *, not char *

Avoids compiler warnings calling gui->prompt_for() with a const char *


---

diff --git a/src/hid.h b/src/hid.h
index dd34b8e..94734ba 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -468,7 +468,7 @@ typedef enum
     /* Prompts the user to enter a string, returns the string.  If
        default_string isn't NULL, the form is pre-filled with this
        value.  "msg" is like "Enter value:".  */
-    char *(*prompt_for) (char *msg_, char *default_string_);
+    char *(*prompt_for) (const char *msg_, const char *default_string_);
 
     /* Prompts the user for a filename or directory name.  For GUI
        HID's this would mean a file select dialog box.  The 'flags'
diff --git a/src/hid/batch/batch.c b/src/hid/batch/batch.c
index f3b7d3c..7db8a54 100644
--- a/src/hid/batch/batch.c
+++ b/src/hid/batch/batch.c
@@ -363,7 +363,7 @@ batch_report_dialog (char *title, char *msg)
 }
 
 static char *
-batch_prompt_for (char *msg, char *default_string)
+batch_prompt_for (const char *msg, const char *default_string)
 {
   static char buf[1024];
   if (default_string)
diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index d5dc9e1..6111ec3 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -287,7 +287,7 @@ nogui_report_dialog (char *title, char *msg)
 }
 
 static char *
-nogui_prompt_for (char *msg, char *default_string)
+nogui_prompt_for (const char *msg, const char *default_string)
 {
   static char buf[1024];
   if (default_string)
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 7d59672..a3947d0 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -867,7 +867,7 @@ ghid_report_dialog (char *title, char *msg)
 }
 
 char *
-ghid_prompt_for (char *msg, char *default_string)
+ghid_prompt_for (const char *msg, const char *default_string)
 {
   char *rv;
 
diff --git a/src/hid/gtk/gui-dialog.c b/src/hid/gtk/gui-dialog.c
index ec91233..1f6b11a 100644
--- a/src/hid/gtk/gui-dialog.c
+++ b/src/hid/gtk/gui-dialog.c
@@ -46,7 +46,7 @@ RCSID ("$Id$");
 
 /* ---------------------------------------------- */
 gchar *
-ghid_dialog_input (gchar * prompt, gchar * initial)
+ghid_dialog_input (const char * prompt, const char * initial)
 {
   GtkWidget *dialog, *vbox, *label, *entry;
   gchar *string;
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 7686106..4ad6cdc 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -332,7 +332,7 @@ int ghid_dialog_close_confirm (void);
 #define GUI_DIALOG_CLOSE_CONFIRM_NOSAVE 1
 #define GUI_DIALOG_CLOSE_CONFIRM_SAVE   2
 gint ghid_dialog_confirm_all (gchar * message);
-gchar *ghid_dialog_input (gchar * prompt, gchar * initial);
+gchar *ghid_dialog_input (const char * prompt, const char * initial);
 void ghid_dialog_about (void);
 
 char * ghid_fileselect (const char *, const char *, char *, char *, const char *, int);
diff --git a/src/hid/lesstif/dialogs.c b/src/hid/lesstif/dialogs.c
index 544a718..b3cc375 100644
--- a/src/hid/lesstif/dialogs.c
+++ b/src/hid/lesstif/dialogs.c
@@ -544,7 +544,7 @@ lesstif_fileselect (const char *title, const char *descr,
 		    const char *history_tag, int flags)
 {
 
-  return lesstif_prompt_for ((char *)title, default_file);
+  return lesstif_prompt_for (title, default_file);
 }
 
 /* ------------------------------------------------------------ */
@@ -553,7 +553,7 @@ static Widget prompt_dialog = 0;
 static Widget prompt_label, prompt_text;
 
 char *
-lesstif_prompt_for (char *msg, char *default_string)
+lesstif_prompt_for (const char *msg, const char *default_string)
 {
   char *rv;
   XmString xs;
@@ -592,7 +592,7 @@ lesstif_prompt_for (char *msg, char *default_string)
   xs = XmStringCreateLocalized ((char *)msg);
   stdarg (XmNlabelString, xs);
   XtSetValues (prompt_label, args, n);
-  XmTextSetString (prompt_text, default_string);
+  XmTextSetString (prompt_text, (char *)default_string);
   XmTextSetCursorPosition (prompt_text, strlen (default_string));
   wait_for_dialog (prompt_dialog);
   rv = XmTextGetString (prompt_text);
diff --git a/src/hid/lesstif/lesstif.h b/src/hid/lesstif/lesstif.h
index 54e4ced..3bc822b 100644
--- a/src/hid/lesstif/lesstif.h
+++ b/src/hid/lesstif/lesstif.h
@@ -50,7 +50,7 @@ extern void lesstif_insert_style_buttons (Widget menu);
 extern void lesstif_styles_update_values ();
 extern void lesstif_update_layer_groups ();
 extern void lesstif_update_status_line ();
-extern char *lesstif_prompt_for (char *, char *);
+extern char *lesstif_prompt_for (const char *, const char *);
 extern char *lesstif_fileselect (const char *, const char *,
 				 char *, char *,
 				 const char *, int);
